astropy/cosmology/parameter/_core.py

```diff
@@ -129,12 +149,14 @@ class Parameter:
         def __init__(
             self,
             *,
+            default: Any | Literal[Sentinel.has_no_default] = HASNODEFAULT,
```

Comment:
I'll remove the type hint before we merge.

```suggestion
            default = HASNODEFAULT,
```